experiments/AMIP/moist_mpi_earth/push_pull.jl

```diff
@@ -154,16 +149,38 @@ function atmos_pull!(cs)
     end
 
     # calculate turbulent fluxes on atmos grid and save in atmos cache
-    info_sfc = (; ice_mask = ice_mask)
     parent(atmos_sim.integrator.p.T_sfc) .= parent(T_sfc_cpl)
-    parent(atmos_sim.integrator.p.ρ_sfc) .= parent(ρ_sfc_cpl)
-    parent(atmos_sim.integrator.p.q_sfc) .= parent(q_sfc_cpl)
-    if :z0b in propertynames(integrator.p.surface_scheme)
-        parent(atmos_sim.integrator.p.z0b) .= parent(z0b_cpl)
-        parent(atmos_sim.integrator.p.z0m) .= parent(z0m_cpl)
+    if :z0b in propertynames(atmos_sim.integrator.p.surface_scheme)
+        parent(atmos_sim.integrator.p.sfc_inputs.z0b) .= parent(z0b_cpl)
+        parent(atmos_sim.integrator.p.sfc_inputs.z0m) .= parent(z0m_cpl)
     end
 
-    calculate_surface_fluxes_atmos_grid!(atmos_sim.integrator, info_sfc)
+    Fields.bycolumn(axes(atmos_sim.integrator.p.ts_sfc)) do colidx
+        ClimaAtmos.set_surface_thermo_state!(
```

Comment:
This is assuming a saturated surface. We will need to:
- 1. re-enable setting `ρ_sfc` from the coupler (@szy21 was there a plan to do this? We can also extend the `surface_thermo_state` function if not. But even for the stand-alone runs you may want to not be bound to surface saturation?). Both land and ice surfaces will rarely be saturated. 
- 2. cap evaporation over ice, as was previously done in the `correct_e_over_ice` function in the coupler. 
To avoid scope creep, let's merge this but address the surface saturation ASAP. #237

Reply:
Do you mean setting `q_sfc` from the coupler? In any case, I don't think there is a plan to do this. We used to have to option to set both `ρ_sfc` and `q_sfc`, but then it was refactored to set `ts_sfc`. In the latest release of ClimaAtmos we have `set_surface_thermo_state!(::Coupled, args...) = nothing`, so I would assume it will use the `ts_sfc` set in the coupler.

Reply:
Yes, and ok, no probs. We'll make our own `set_surface_thermo_state!` extension. Thanks!